packages/react-native-gesture-handler/src/web/handlers/GestureHandler.ts

```diff
@@ -383,21 +385,34 @@ export default abstract class GestureHandler implements IGestureHandler {
       return;
     }
     this.ensurePropsRef();
-    const { onGestureHandlerEvent, onGestureHandlerTouchEvent }: PropsRef =
-      this.propsRef!.current;
+    const {
+      onGestureHandlerEvent,
+      onGestureHandlerTouchEvent,
+      onGestureHandlerReanimatedEvent,
+      onGestureHandlerReanimatedTouchEvent,
+    }: PropsRef = this.propsRef!.current;
 
     const touchEvent: ResultEvent<GestureTouchEvent> | undefined =
       this.transformTouchEvent(event);
 
-    if (touchEvent) {
-      if (
-        onGestureHandlerTouchEvent &&
-        this.actionType === ActionType.NATIVE_DETECTOR
-      ) {
-        invokeNullableMethod(onGestureHandlerTouchEvent, touchEvent);
-      } else {
-        invokeNullableMethod(onGestureHandlerEvent, touchEvent);
-      }
+    if (!touchEvent) {
+      return;
+    }
+
+    if (this.actionType === ActionType.NATIVE_DETECTOR) {
+      invokeNullableMethod(
+        this.forReanimated
+          ? onGestureHandlerReanimatedTouchEvent!
+          : onGestureHandlerTouchEvent,
+        touchEvent
+      );
+    } else {
+      invokeNullableMethod(
+        this.forReanimated
+          ? onGestureHandlerReanimatedEvent!
+          : onGestureHandlerEvent,
+        touchEvent
+      );
```

Comment:
Correct me if I'm wrong, but when we do not use V3, it means that `forReanimated` is `false`. In that case, why do we have

```js
invokeNullableMethod(
        this.forReanimated
          ? onGestureHandlerReanimatedEvent!
          : onGestureHandlerEvent,
        touchEvent
      );
```

instead of simply

```js
invokeNullableMethod(onGestureHandlerEvent touchEvent);
```